collections/lists.go

```diff
@@ -22,3 +22,29 @@ func RemoveElementFromList(list []string, element string) []string {
 	return out
 }
 
+// MakeCopyOfList will return a new list that is a copy of the given list.
+func MakeCopyOfList(list []string) []string {
+	copyOfList := make([]string, len(list))
+	copy(copyOfList, list)
+	return copyOfList
+}
+
+// BatchListIntoGroupsOf will group the provided string slice into groups of size n, with the last of being truncated to
+// the remaining count of strings.  Returns nil if n is <= 0
+func BatchListIntoGroupsOf(slice []string, batchSize int) [][]string {
+	if batchSize <= 0 {
+		return nil
+	}
+
+	// Taken from SliceTricks: https://github.com/golang/go/wiki/SliceTricks#batching-with-minimal-allocation
+	// Intuition: We repeatedly slice off batchSize elements from slice and append it to the output, until there
```

Comment:
Hah, I didn't realize they had a batch example right in there 👍

Reply:
Yup. Saves me the trouble :)